scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,134 @@
 
 Typical usage example::
 
-    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml \
-        --species_set default
+    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml --species_set name \
+    --host mysql-ens-compara-prod-X --port XXXX --user ensro --out_dir /path/to/out/dir
 
 """
 
 import argparse
+import os
+import subprocess
+from typing import Dict, List
+
+import numpy
+from sqlalchemy import create_engine
 
 from ensembl.compara.config import get_species_set_by_name
 
 
-def dump_genomes():
-    """Docstring"""
+def dump_genomes(species_list: List[str], species_set_name: str, host: str, port: int,
+                 user: str, out_dir: str) -> None:
+    """Dumps canonical peptides of protein-coding genes for a specified list of species.
+
+    Peptides are dumped from the latest available core databases to FASTA files
+    using `dump_gene_set_from_core.pl`.
+
+    Args:
+        species_list: A list of species (genome) names to dump.
+        species_set_name: Species set (collection) name.
+        host: Database host.
+        port: Host port.
+        user: Server username.
+        out_dir: Directory to place `species_set_name/core_name.fasta` dumps.
+
+    Raises:
+        FileExistsError: If directory `out_dir/species_set_name` already exists.
+        sqlalchemy.exc.OperationalError: If `user` cannot read from `host:port`.
+        RuntimeError: If no cores are found for any species in `species_list` (`species_set_name`)
+            on `host:port`.
+
+    """
+    cores = get_core_names(species_list, host, port, user)
+    dump_cores = [core for species, core in cores.items() if core != ""]
+
+    if len(dump_cores) == 0:
+        raise RuntimeError(f"No cores found for the species set '{species_set_name}' on the specified host.")
+
+    dumps_dir = os.path.join(out_dir, species_set_name)
+    os.mkdir(dumps_dir)
+
+    script = os.path.join(os.environ["ENSEMBL_ROOT_DIR"], "ensembl-compara", "scripts", "dumps",
+                          "dump_gene_set_from_core.pl")
+
+    for core in dump_cores:
+        out_file = os.path.join(dumps_dir, f"{core}.fasta")
+        subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
+                       "-outfile", out_file], capture_output=True, check=True)
+
+
+def find_latest_core(core_names: List[str]) -> str:
+    """Returns the latest core database among the available ones.
+
```

Comment:
Just in case, it may be worth adding the expected format of the core database names?
```suggest
    The core database names are expected in the format:
    ``<species-name>_core_(<eg-rel>_)<ensembl-rel>_<version>``, e.g. ``erinaceus_europaeus_core_106_1``.
```